### ISSUE TITLE
External users (B2B) not able to login

### ISSUE THREAD

**Issue by kishoresahas** [wpo365/wpo365-login#13]
External users are not able to login.
For eg : I have registered with sample.com AAD.

Not working for external users, who are invited with the below setup. 
https://docs.microsoft.com/en-us/azure/active-directory/active-directory-b2b-what-is-azure-ad-b2b

**Comment by wpo365** [comment]
I haven't been able to test this thus far, but I'm quite sure that this requires additional configuration in your Azure AD so that external users have access to the App you registered for this. See "Admins adding guest users to an application" here (link)[https://support.office.com/en-us/article/Enable-or-disable-auto-acceleration-for-your-SharePoint-Online-tenancy-74985ebf-39e1-4c59-a74a-dcdfd678ef83]

**Comment by wpo365** [comment]
No news is good news so I'm closing this issue. Feel free to reopen it if the issue is not resolved.

**Comment by kishoresahas** [comment]
That didnt solved my issue ..
Are you using Graph Api ? I go this below reference ..
https://stackoverflow.com/questions/42095475/graph-api-requests-for-guest-users-in-azure-ad

**Comment by wpo365** [comment]
@kishoresahas If you go to the Admin Dashboard and open the WPO365-Options page then you'll find a Miscellaneous section. Here you can check "Debug". This will write a lot of information in the wp-content\wp-debug.log file (this file may also be created somewhere else, depending on your Wordpress configuration). When you repeat your steps on more time in Debugging mode and post the logfile here or send it to info@wpo365.com (it may contain sensitive information so maybe you want to clean it first a bit) then I can have a look at it. 

There may be an issue with external users to get their id token, but seeing the log files would help. Otherwise it would take some time to set up a similar test environment.

**Comment by kishoresahas** [comment]
I have send the log file to you email [info@wpo365.com](mail:info@wpo365.com)

**Comment by wpo365** [comment]
@kishoresahas I noticed a few things. First of all, can you please update the plugin to the latest version. I can't tell what version you use, but in a somewhat older version there was a problem with setting cookies but this should now be resolved.

Then I see that for at least one user (with Wordpress id #2) it seems to work fine. But then I guess you try to logon with an external user and you receive a response from Microsoft stating that "Invalid resource. The client has requested access to a resource which is not listed in the requested permissions in the client\'s application registration."

Basically, what the plugin does is to ask Azure AD to logon to itself. It will take the application id of the Enterprise App that you registered. If it works for "internal" users but not for "guest" users, then I suspect that in Azure Active Directory a configuration is missing that allows "guests" to logon to your Enterprise App.

**Comment by wpo365** [comment]
My guess is that you need to add some additional configuration when you go portal.azure.com and then click "Azure Active Directory" > "Enterprise Application" > "All Application" and then click the application you created for the WPO365-login plugin (as described here https://www.wpo365.com/how-to-install-wordpress-office-365-login-plugin/). Maybe this link is helpful https://docs.microsoft.com/en-us/azure/active-directory/application-sign-in-problem-federated-sso-gallery even though it not necessarily covers Azure B2B.

**Comment by kishoresahas** [comment]
Quick update on this: Problem will be solved with the latest version 2.3.
